templates/commands/render.go

```diff
@@ -235,24 +278,167 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 		return err
 	}
 
-	_ = templateDir // TODO: add template rendering logic
+	spec, err := loadSpecFile(rp.fs, templateDir, r.flagSpec)
+	if err != nil {
+		return err
+	}
 
+	scratchDir, err := rp.tempDirNamer(scratchDirNamePart)
+	if err != nil {
+		return err
+	}
+	if err := rp.fs.MkdirAll(scratchDir, 0o700); err != nil {
```

Comment:
Wondering should file permissions also be constant?

Reply:
Done.